pkg/codegen/dotnet/doc.go

```diff
@@ -72,17 +72,13 @@ func (d DocLanguageHelper) GetDocLinkForFunctionInputOrOutputType(pkg *schema.Pa
 }
 
 // GetLanguageTypeString returns the DotNet-specific type given a Pulumi schema type.
-func (d DocLanguageHelper) GetLanguageTypeString(pkg *schema.Package, moduleName string, t schema.Type, input bool) string {
-	info, ok := pkg.Language["csharp"].(CSharpPackageInfo)
-	if !ok {
-		info = CSharpPackageInfo{}
-	}
-	typeDetails := map[*schema.ObjectType]*typeDetails{}
+func (d DocLanguageHelper) GetTypeName(pkg *schema.Package, t schema.Type, input bool, relativeToModule string) string {
+	info, _ := pkg.Language["csharp"].(CSharpPackageInfo)
```

Comment:
Is this guaranteed to succeed now? How do we know? If so, can we document the contract, wherever it is?

Reply:
No, it isn't. If it doesn't succeed, then we use the zero value of `CSharpPackageInfo` as our default.